h2o-core/src/main/java/hex/ModelMetricsRegressionCoxPH.java

```diff
@@ -147,83 +134,455 @@ long discordant() {
         return ntotals - nconcordant - nties;
       }
 
+      @Override
+      public String toString() {
+        return "Stats{" +
+                "ntotals=" + ntotals +
+                ", nconcordant=" + nconcordant +
+                ", ndiscordant=" + discordant() +
+                ", nties=" + nties +
+                '}';
+      }
+
       Stats plus(Stats s2) {
-        return new Stats(ntotals + s2.ntotals, nNotNaN + s2.nNotNaN, nconcordant + s2.nconcordant, nties + s2.nties);
+        return new Stats(ntotals + s2.ntotals, nconcordant + s2.nconcordant, nties + s2.nties);
       }
     }
     
     static Stats concordance(final Vec startVec, final Vec stopVec, final Vec eventVec, List<Vec> strataVecs, final Vec estimateVec) {
-      final long length = estimateVec.length();
+      final Vec durations = durations(startVec, stopVec);
+      Frame fr = prepareFrameForConcordanceComputation(eventVec, strataVecs, estimateVec, durations);
+      return concordanceStats(fr);
+    }
+
+    private static Frame prepareFrameForConcordanceComputation(Vec eventVec, List<Vec> strataVecs, Vec estimateVec, Vec durations) {
+      final Frame fr = new Frame();
+      fr.add("duration", durations);
+      fr.add("event", eventVec);
+      fr.add("estimate", estimateVec);
+      for (int i = 0; i < strataVecs.size(); i++) {
+        fr.add("strata_" + i, strataVecs.get(i));
+      }
+      return fr;
+    }
+
+    private static Vec durations(Vec startVec, Vec stopVec) {
+      Vec vec = null == startVec ? stopVec.makeZero() : startVec;
```

Comment:
I don't see the temporary frame disposed of anywhere.

On another note - if startVec is not given then durations = startVec. You can skip the MRTask altogether.

Reply:
updated